### PR TITLE
Migrate JupterLab from 2.x to 3.x

### PR DESCRIPTION
**Pull request by NikitaGandhi**
- JupyterLab 3.x comes with many inbuilt features - https://blog.jupyter.org/jupyterlab-3-0-is-out-4f58385e25bb
- JupyterLab releases prebuilt extensions and available to PyPI - reduce step to install JupyterLab extensions and rebuild  JupyterLab

### CODE REVIEW

**Comment by NikitaGandhi** [comment]
This PR requires extended testing before updating sandbox environment and not ready to merge :)

File: docker/constraints-odc.txt [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+odc_aws>=0.2.1.dev
+odc_geom>=0.2.1.dev
+odc_io>=0.2.1.dev
+odc_aio>=0.2.1.dev
+odc_ppt>=0.2.1.dev
+odc_dscache>=0.2.1.dev
+odc_dtools>=0.2.1.dev
+odc_stats>=0.2.1.dev
+odc-apps-dc-tools>=0.2.1.dev
+odc-apps-cloud>=0.2.1.dev
```
Comment (alexgleith) [inline docker/constraints-odc.txt]: These libraries are all in flux right now, we may want to wait for them to be in PyPI and pull from there... Or deal with that change later!

**Review by alexgleith — APPROVED** [review]
An enormous list of library changes. Comprehensive and awesome, Nikita!

**Comment by omad** [comment]
Is it possible to roll this out in parallel with the existing version 2.x?

It'd be great to be able to test without potentially breaking things for everyone.

**Comment by alexgleith** [comment]
Do you mean we maintain two different image builds, @omad? If so, then I'm not sure that it's a good idea to maintain two environments, as we struggle enough maintaining one!

We have got this image deployed in parallel, though, so they can both be tested.

**Comment by omad** [comment]
> Do you mean we maintain two different image builds, @omad? If so, then I'm not sure that it's a good idea to maintain two environments, as we struggle enough maintaining one!

Nhooo, definitely do not want that.

> We have got this image deployed in parallel, though, so they can both be tested.

Awesome, that's what I was hoping for, and is what's needed for testing and approving this kind of change. :D

**Comment by robbibt** [comment]
Hey @NikitaGandhi, a silly question: where can I actually test this? Is it deployed on a Sandbox somewhere?

**Comment by cbur24** [comment]
Something weird I'm noticing...when I first open a notebook and use "restart kernel and run all cells", some of the cells in the middle of the notebook are skipped (not evaluated). If I clear the kernel and do it again, it works fine.

image

**Comment by cbur24** [comment]
we still have this hacky memory counter in the bottom left of the window, I think we should remove it given we now have the graphic in the top-left corner

image

**Comment by cedricump** [comment]
Can we ensure that the French language pack is install by default. so user can readily access the language through JupyterLab settings without pip installing it? the latest version is 100% translated. Tested on dev through pip install and it is working.

**Comment by NikitaGandhi** [comment]
> we still have this hacky memory counter in the bottom left of the window, I think we should remove it given we now have the graphic in the top-left corner
> 
> image

- Updated top-bar extention to include `CPU usage`, `Memory usage`, `Logout button` and `theme-toggle button` and removed `jupyterlab-topbar-text` ("Hi there" text on topbar):

image

- Not able to remove/disable memory usage metrics from bottom as this settings comes from `jupyter-resource-usage` extension as default and also required by top-bar `jupyterlab-system-monitor` extension. But the bottom status-bar is enable/disable from menu `view -> Show status bar` by user as an option.

File: docker/constraints-odc.txt [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-odc_ppt==0.2.1.dev2305
-odc_dscache==0.2.1.dev2305
-odc_dtools==0.2.1.dev2305
-odc_stats==0.2.1.dev2305
-odc-apps-dc-tools==0.2.1.dev2305
-odc-apps-cloud==0.2.1.dev2305
\ No newline at end of file
+odc-aio==0.2.1.dev2476
+odc-cloud==0.2.1.dev2644
+odc-index==0.2.1.dev2420
```
Comment (Kirill888) [inline docker/constraints-odc.txt]: no such thing anymore, it was renamed into `odc-stac`

File: docker/constraints-odc.txt [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+odc-aws==0.2.1.dev2476
+odc-ppt==0.2.1.dev2400
+odc-dtools==0.2.1.dev2400
```
Comment (Kirill888) [inline docker/constraints-odc.txt]: no such thing anymore, those got merged/deleted `odc-cloud`.

File: docker/constraints-odc.txt [hunk 4]
```diff
@@ -1,21 +1,10 @@
...
 datacube==1.8.5
 odc-algo==0.2.0a3
-odc-apps-cloud==0.2.0a1
-odc-apps-dc-tools==0.2.0a1
+odc-cloud==0.2.0a0
 odc-dscache==0.2.0a1
-odc-geom==0.2.0a0
 odc-io==0.2.0a0
-odc_stats==0.2.0a2
+odc-stac==0.2.0a4
```
Comment (Kirill888) [inline docker/constraints-odc.txt]: can we please bump this one:
```suggestion
odc-stac==0.2.0a5
```

**Comment by NikitaGandhi** [comment]
> Can we ensure that the French language pack is install by default. so user can readily access the language through JupyterLab settings without pip installing it? the latest version is 100% translated. Tested on dev through pip install and it is working.

added

**Review by mmochan — APPROVED** [review]
LGTM!

**Pull request merged**